src/shadowbox/server/manager_service.ts

```diff
@@ -147,6 +149,37 @@ export class ShadowsocksManagerService {
     next();
   }
 
+  // Changes the server's hostname.  Hostname must be a valid domain or IP address
+  public changeHostname(req: RequestType, res: ResponseType, next: restify.Next): void {
+    logging.debug(`changeHostname request: ${JSON.stringify(req.params)}`);
+
+    const hostname = req.params.hostname;
+    if (typeof hostname === 'undefined') {
```

Comment:
nit: `if (!hostname)` to capture `null`

Reply:
We capture that in the type guard.  This way is slightly more precise in that we give an error that an empty string is an invalid hostname instead of missing